pkg/db/migrations/202311151850_add_resources.go

```diff
@@ -10,10 +10,11 @@ import (
 func addResources() *gormigrate.Migration {
 	type Resource struct {
 		Model
-		ConsumerID string         `gorm:"index"`
-		Version    int            `gorm:"not null"`
-		Manifest   datatypes.JSON `gorm:"type:json"`
-		Status     datatypes.JSON `gorm:"type:json"`
+		ConsumerID         string         `gorm:"index"`
+		Version            int            `gorm:"not null"`
+		ObservedGeneration int            `gorm:"not null"`
```

Comment:
do we need this tag('not null') for `ObservedGeneration`? when a spec is created by user, before the status is returned rom work-agent, the `ObservedGeneration` should be empty?